pkg/util/machine_image_test.go

```diff
@@ -1,11 +1,77 @@
 package util
 
 import (
+	"time"
+
 	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/gardener/test-infra/pkg/common"
 )
 
+const (
+	imageName  = "gardenlinux"
+	arch_amd64 = "amd64"
+	arch_arm64 = "arm64"
+)
+
+var _ = Describe("get machine images from a cloudprofile", func() {
+	var (
+		cloudprofile gardencorev1beta1.CloudProfile
+		expiredTime  metav1.Time
+		futureTime   metav1.Time
+	)
+
+	BeforeEach(func() {
+		expiredTime = metav1.NewTime(time.Now())
+		futureTime = metav1.NewTime(time.Now().Add(time.Hour * 24))
+		cloudprofile = gardencorev1beta1.CloudProfile{
+			Spec: gardencorev1beta1.CloudProfileSpec{
+				MachineImages: []gardencorev1beta1.MachineImage{
+					{
+						Name: imageName,
+						Versions: []gardencorev1beta1.MachineImageVersion{
+							{ExpirableVersion: gardencorev1beta1.ExpirableVersion{
+								Version:        "3.4.5",
+								ExpirationDate: &futureTime,
+							}, Architectures: []string{arch_amd64, arch_arm64}},
+							{ExpirableVersion: gardencorev1beta1.ExpirableVersion{
+								Version:        "2.3.4",
```

Comment:
Maybe also add a version `2.3.3` so that we test that n-1 would in fact return `2.3.4` instead of `2.3.3`?

Reply:
https://github.com/gardener/test-infra/pull/525/commits/d03e116b02fe86dedfab749594d0afc4215ac45a